### ISSUE TITLE
C++ backend 2-d array struct name collison

### ISSUE THREAD

**Issue by lucifer1004** [lfortran/lfortran#747]
```fortran
program test
   integer :: a(11, 11), b(111, 1)
   a = 1
   b = 2
end program
```

translates to the following C++ code

```c++
#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <complex>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}


struct dimension_descriptor
{
    int32_t lower_bound, length;
};

struct i32_1111_2
{
    Kokkos::View<int32_t*>* data;
    dimension_descriptor dims[2];
    bool is_allocated;

    i32_1111_2(Kokkos::View<int32_t*>* data_): data{data_} {};
};


struct i32_1111_2
{
    Kokkos::View<int32_t*>* data;
    dimension_descriptor dims[2];
    bool is_allocated;

    i32_1111_2(Kokkos::View<int32_t*>* data_): data{data_} {};
};

// Forward declarations
namespace {
}

// Implementations
namespace {

void main2() {
    Kokkos::View<int32_t*> a_data("a_data", 121);
    i32_1111_2 a_value(&a_data);
    i32_1111_2* a = &a_value;
    a->dims[0].lower_bound = 1;
    a->dims[0].length = 11;
    a->dims[1].lower_bound = 1;
    a->dims[1].length = 11;
    Kokkos::View<int32_t*> b_data("b_data", 111);
    i32_1111_2 b_value(&b_data);
    i32_1111_2* b = &b_value;
    b->dims[0].lower_bound = 1;
    b->dims[0].length = 111;
    b->dims[1].lower_bound = 1;
    b->dims[1].length = 1;
    a->data = 1;
    b->data = 2;
}

}
int main(int argc, char* argv[])
{
    Kokkos::initialize(argc, argv);
    main2();
    Kokkos::finalize();
    return 0;
}
```

As can be seen, both `(11, 11)` and `(111,1)` arrays are named as `i32_1111_2`, causing a collision.

This should be easy to fix.

**Comment by certik** [comment]
Nice catch and fix! Thank you.

**Comment by lucifer1004** [comment]
Closed by #748
